apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2.Tests/RetryingReadRowsTest.cs

```diff
@@ -0,0 +1,450 @@
+﻿// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.V2.Tests
+{
+    public class RetryingReadRowsTest
+    {
+        [Fact]
+        public async Task SimpleRetryWithKeys()
+        {
+            var request = new ReadRowsRequest
+            {
+                Rows = RowSet.FromRowKeys("a", "b", "c")
+            };
+            var client = Utilities.CreateReadRowsMockClient(
+                request,
+                initialStreamResponse: new[]
+                {
+                    new ReadRowsResponse
+                    {
+                        Chunks =
+                        {
+                            CreateChunk("a", "cf1", "column1", "value1", commitRow: true)
+                        }
+                    }
+                },
+                responsesForRetryStreams: new[]
+                {
+                    new []
+                    {
+                        new ReadRowsResponse
+                        {
+                            Chunks =
+                            {
+                                CreateChunk("b", "cf1", "column2", "value2", commitRow: true)
+                            }
+                        }
+                    }
+                });
+
+            var stream = client.ReadRows(request);
+
+            int rowCount = 0;
```

Comment:
Do we have any guarantee about the order in which the rows should be returned?

Reply:
Yes, they are returned in lexicographical order. Are you worried that this test can be fooled by us incorrectly duplicating a row? I'll update it.

Reply:
No, it was more that it feels like we could just all the results to a list and assert how they looked that way - this feels a little more roundabout to me.

Reply:
Yup, that's definitely better. Done.